packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdOffline.spec.ts

```diff
@@ -0,0 +1,363 @@
+
+		// Get Pending state and close
+		assert(container2.closeAndGetPendingLocalState !== undefined, "Missing method!");
+		const pendingState = await container2.closeAndGetPendingLocalState();
+
+		// Load from the pending state
+		const container3 = await provider.loadContainer(
+			runtimeFactory,
+			{ configProvider },
+			undefined,
+			pendingState,
+		);
+		container3.disconnect();
+
+		// Testing the get snapshot call
+		const mainObject3 = (await container3.getEntryPoint()) as TestDataObject;
+		const runtime3 = mainObject3.containerRuntime;
+		assert(runtime3.storage.getSnapshot !== undefined, "getSnapshot should be defined");
+
+		// Try to load the data stores with groupIds
+		const handleA3 = mainObject3._root.get<IFluidHandle<TestDataObject>>("dataObjectA");
+		const handleB3 = mainObject3._root.get<IFluidHandle<TestDataObject>>("dataObjectB");
+		assert(handleA3 !== undefined, "handleA3 should not be undefined");
+		assert(handleB3 !== undefined, "handleB3 should not be undefined");
+
+		// loading group call
+		const dataObjectA3 = await handleA3.get();
+		const dataObjectB3 = await handleB3.get();
+		assert.equal(dataObjectA3._root.get("A"), "A", "A should be set");
+		assert.equal(dataObjectA3._root.get("A2"), "A2", "A2 should be set");
+		assert.equal(dataObjectB3._root.get("B"), "B", "B should be set");
+		container3.connect();
+		await provider.ensureSynchronized();
+	});
+
+	it("GroupId offline with refresh", async () => {
+		if (provider.driver.type !== "local") {
+			return;
+		}
+		// Load basic container stuff
+		const container = (await provider.createContainer(runtimeFactory, {
+			configProvider,
+		})) as IContainerExperimental;
+		const mainObject = (await container.getEntryPoint()) as TestDataObject;
+		const containerRuntime = mainObject.containerRuntime;
+
+		// Create data stores with loadingGroupIds
+		const dataStoreA = await containerRuntime.createDataStore(
+			testDataObjectType,
+			loadingGroupId,
+		);
+		const dataStoreB = await containerRuntime.createDataStore(
+			testDataObjectType,
+			loadingGroupId,
+		);
+
+		// Attach the data stores
+		const dataObjectA = (await dataStoreA.entryPoint.get()) as TestDataObject;
+		const dataObjectB = (await dataStoreB.entryPoint.get()) as TestDataObject;
+		mainObject._root.set("dataObjectA", dataObjectA.handle);
+		mainObject._root.set("dataObjectB", dataObjectB.handle);
+		dataObjectA._root.set("A", "A");
+		dataObjectB._root.set("B", "B");
+		await provider.ensureSynchronized();
+
+		const { container: summarizingContainer, summarizer } = await createSummarizerFromFactory(
+			provider,
+			container,
+			dataObjectFactory,
+			undefined,
+			undefined,
+			undefined,
+			undefined,
+			configProvider,
+		);
+
+		await provider.ensureSynchronized();
+		const { summaryVersion, summaryRefSeq } = await summarizeNow(summarizer);
+		await provider.ensureSynchronized();
+
+		// Refresh snapshot - this to validate that we built a system that would work with refresh
```

Comment:
This is the start of where I hack a "refresh", obviously this should be updated once we have refresh.